pkg/volume/csi/csi_attacher.go

```diff
@@ -229,12 +235,167 @@ func (c *csiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.No
 }
 
 func (c *csiAttacher) GetDeviceMountPath(spec *volume.Spec) (string, error) {
-	glog.V(4).Info(log("attacher.GetDeviceMountPath is not implemented"))
-	return "", nil
+	glog.V(4).Info(log("attacher.GetDeviceMountPath(%v)", spec))
+
+	if spec == nil {
+		return "", fmt.Errorf("attacher.GetDeviceMountPath failed, spec is nil")
+	}
+
+	csiSource, err := getCSISourceFromSpec(spec)
+	if err != nil {
+		glog.Error(log("attacher.GetDeviceMountPath failed getting CSI Source from Spec: %v", err))
+		return "", err
+	}
+	if csiSource.VolumeHandle == "" {
+		return "", fmt.Errorf("attacher.GetDeviceMountPath failed, volume handle empty")
+	}
+	if csiSource.Driver == "" {
+		return "", fmt.Errorf("attacher.GetDeviceMountPath failed, driver name empty")
+	}
+
+	deviceMountPath := path.Join(c.plugin.host.GetPluginDir(csiSource.Driver), "mounts", csiSource.VolumeHandle)
+	glog.V(4).Infof("attacher.GetDeviceMountPath succeeded, deviceMountPath: %s", deviceMountPath)
+	return deviceMountPath, nil
 }
 
 func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMountPath string) error {
-	glog.V(4).Info(log("attacher.MountDevice is not implemented"))
+	glog.V(4).Infof(log("attacher.MountDevice(%s, %s)", devicePath, deviceMountPath))
+
+	// Setup
+	ctx, cancel := grpctx.WithTimeout(grpctx.Background(), csiTimeout)
```

Comment:
Move creating `ctx` down to right before you actually use it for the first time. No need to waste cycles if you end up not using it.